src/context/WalletProvider.tsx

```diff
@@ -702,12 +716,20 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
         } catch (err: any) {
           if (err.code === ERRCODE.UserRejected) {
             console.warn('[Wallet] User rejected the request')
-            throw err // return false // todo: sendTx reject => false
+            throw new RejectRequestError()
           }
           throw err
         }
       } else if (isCosmosWallet(currentState)) {
-        return await executeCosmosTransaction(transaction as CosmosTransaction, currentState.provider)
+        try {
+          return await executeCosmosTransaction(transaction as CosmosTransaction, currentState.provider)
+        } catch (err: any) {
+          if (err?.message === 'Request rejected') {
+            console.warn('[Wallet] User rejected the request')
+            throw new RejectRequestError()
```

Comment:
We need to think a little bit about this code behavior, since the rejection of a transaction by a user should not throw an error. That's exactly what was written in the comment you deleted.